uxarray/grid/utils.py

```diff
@@ -58,3 +59,65 @@ def _replace_fill_values(grid_var, original_fill, new_fill, new_dtype=None):
     grid_var[fill_val_idx] = new_fill
 
     return grid_var
+
+
+def _fmms(a, b, c, d):
+    """
+    Calculate the difference of products using the FMA (fused multiply-add) operation: (a * b) - (c * d).
+
+    This operation leverages the fused multiply-add operation when available on the system and rounds the result only once.
+    The relative error of this operation is bounded by 1.5 ulps when no overflow and underflow occur.
+
+    Parameters
+    ----------
+    a (float): The first value of the first product.
+    b (float): The second value of the first product.
+    c (float): The first value of the second product.
+    d (float): The second value of the second product.
+
+    Returns
+    -------
+    float: The difference of the two products.
+
+    Example
+    -------
+    >>> _fmms(3.0,2.0,1.0,1.0)
+    5.0
+
+    Reference
+    ---------
+    Claude-Pierre Jeannerod, Nicolas Louvet, and Jean-Michel Muller, Further
+    analysis of Kahan’s algorithm for the accurate computation of 2 x 2 determinants,
+    Mathematics of Computation, vol. 82, no. 284, pp. 2245-2264, 2013.
+    [Read more](https://ens-lyon.hal.science/ensl-00649347) (DOI: 10.1090/S0025-5718-2013-02679-8)
+    """
+    cd = c * d
+    err = pyfma.fma(-c, d, cd)
+    dop = pyfma.fma(a, b, -cd)
+    return dop + err
+
+
+def cross_fma(v1, v2):
```

Comment:
Also add this to internal API

Reply:
I added this as the user_API at the end since this `cross_fma` can provide a way more robust cross product than the `np.cross` and hence highly useful for the geometry operation.